### PR TITLE
Use ResxSourceGenerator for resource generation

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

File: StyleCop.Analyzers/StyleCop.Analyzers/Lightup/.generated/Microsoft.CodeAnalysis.ResxSourceGenerator.CSharp/Microsoft.CodeAnalysis.ResxSourceGenerator.CSharp.CSharpResxGenerator/StyleCop.Analyzers.DocumentationRules.DocumentationResources.Designer.cs [hunk 1]
```diff
@@ -0,0 +1,406 @@
+﻿// <auto-generated/>
+
+#nullable enable
+using System.Reflection;
+
+
+namespace StyleCop.Analyzers.DocumentationRules
+{
+    internal static partial class DocumentationResources
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/Lightup/.generated/Microsoft.CodeAnalysis.ResxSourceGenerator.CSharp/Microsoft.CodeAnalysis.ResxSourceGenerator.CSharp.CSharpResxGenerator/StyleCop.Analyzers.DocumentationRules.DocumentationResources.Designer.cs:9]: 💭 I'm leaning towards removing the generated output of ResxSourceGenerator from source control

File: StyleCop.Analyzers/Directory.Build.targets [hunk 2]
```diff
@@ -1,17 +1,21 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project>
 
+  <PropertyGroup>
+    <SkipAddAllResxFilesAsAdditionalFiles>true</SkipAddAllResxFilesAsAdditionalFiles>
```
Comment (sharwell) [inline StyleCop.Analyzers/Directory.Build.targets:5]: 📝 I wasn't able to get this working under the default configuration, which is a bad sign for usability

File: StyleCop.Analyzers/Directory.Build.targets [hunk 3]
```diff
@@ -1,17 +1,21 @@
...
+  </PropertyGroup>
+
   <ItemGroup>
-    <!-- Enable code generation for resource files -->
+    <!-- Enable code generation for resource files. -->
     <CodeGenerationEmbeddedResource Include="@(EmbeddedResource)" Exclude="**\*.??-??.resx" />
-    <EmbeddedResource Update="@(CodeGenerationEmbeddedResource)" Generator="ResXFileCodeGenerator" />
+    <AdditionalFiles Include="@(CodeGenerationEmbeddedResource)" GenerateSource="true" />
+
+    <NonCodeGenerationEmbeddedResource Include="@(EmbeddedResource)" Exclude="@(CodeGenerationEmbeddedResource)" />
```
Comment (sharwell) [inline StyleCop.Analyzers/Directory.Build.targets:13]: 💭 Seems like this should be included in the default behavior of ResxSourceGenerator

File: StyleCop.Analyzers/Directory.Build.targets (hunk 2 above)
Comment (sharwell) [reply StyleCop.Analyzers/Directory.Build.targets]: ➡️ This is finally not needed with the recent version of the source generator

File: StyleCop.Analyzers/Directory.Build.targets [hunk 4]
```diff
@@ -2,16 +2,16 @@
 <Project>
 
   <ItemGroup>
-    <!-- Enable code generation for resource files -->
+    <!-- Enable code generation for resource files. -->
     <CodeGenerationEmbeddedResource Include="@(EmbeddedResource)" Exclude="**\*.??-??.resx" />
```
Comment (AArnott) [inline StyleCop.Analyzers/Directory.Build.targets:6]: tip: resource files can also just end with `*.??.resx`. In my revision based on yours, I ended up with this:

```xml
	<ItemGroup>
		<!-- Enable code generation for resource files. -->
		<CodeGenerationEmbeddedResource Include="@(EmbeddedResource)" Exclude="**\*.??-??.resx;**\*.??.resx" />
		<EmbeddedResource Update="@(CodeGenerationEmbeddedResource)" />
		<AdditionalFiles Include="@(CodeGenerationEmbeddedResource)" />

		<NonCodeGenerationEmbeddedResource Include="@(EmbeddedResource)" Exclude="@(CodeGenerationEmbeddedResource)" />
		<EmbeddedResource Update="@(NonCodeGenerationEmbeddedResource)" />
	</ItemGroup>

	<!-- Special handling for embedded resources to show as nested in Solution Explorer -->
	<ItemGroup>
		<!-- Localized embedded resources are just dependent on the parent RESX -->
		<EmbeddedResource Update="**\*.??-??.resx" DependentUpon="$([System.IO.Path]::ChangeExtension($([System.IO.Path]::GetFileNameWithoutExtension(%(Identity))), '.resx'))" />
		<EmbeddedResource Update="**\*.??.resx" DependentUpon="$([System.IO.Path]::ChangeExtension($([System.IO.Path]::GetFileNameWithoutExtension(%(Identity))), '.resx'))" />
	</ItemGroup>
```

Comment (sharwell) [reply StyleCop.Analyzers/Directory.Build.targets:6]: This entire block is going to go away once the ResxSourceGenerator update merges.

File: StyleCop.Analyzers/Directory.Build.targets (hunk 3 above)
Comment (sharwell) [reply StyleCop.Analyzers/Directory.Build.targets]: ➡️ This is now the default behavior

**Pull request merged**